### PR TITLE
[MM-46692] Channel group member count adjustments

### PR DESCRIPTION
**Pull request by BenCookie95**
#### Summary
No longer invoking `/api/v4/channels/:channel_id/member_counts_by_group` on every channel and thread load. Instead we will use the search endpoint as we type in order to return the group member count for that channel. There is an edge case where we will call this endpoint though, if you have a draft for a channel there is a chance you won't have the group in your redux state. If the draft contains more than one @ mention, we use `/api/v4/channels/:channel_id/member_counts_by_group`.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-46692

#### Related Pull Requests

- Has server changes (please link here)
- Has mobile changes (please link here)

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Review by plant99 — APPROVED** [review]

**Comment by furqanmlk** [comment]
HI @BenCookie95, Can you please fix conflict?

**Review by furqanmlk — COMMENTED** [review]
.

**Comment by mkraft** [comment]
@BenCookie95 Gentle ping re conflict resolution to unblock Malik.

File: actions/views/group.js
Comment (BenCookie95) [reply actions/views/group.js:31]: Yeah we will for the popover

**Review by furqanmlk — APPROVED** [review]
Working fine

**Comment by BenCookie95** [comment]
@mkraft This just needs your approval and we can merge! Thanks

**Pull request merged**
